assets/src/apps/authoring/store/groups/layouts/deck/actions/updateActivityRules.ts

```diff
@@ -110,6 +113,12 @@ export const updateActivityRules = createAsyncThunk(
             await updateNestedConditions(conditionsToUpdate, activityTree);
             referencedSequenceIds.push(...findReferencedActivitiesInConditions(conditionsToUpdate));
             referencedVariableKeys.push(...getReferencedKeysInConditions(conditionsToUpdate));
+            referencedSequenceIds.push(
```

Comment:
it's not only here that this gets called, also when the rule is edited in the author.